bezier-rs/lib/src/lib.rs

```diff
@@ -853,6 +838,159 @@ impl Bezier {
 		result
 	}
 
+	/// Approximate a bezier curve with circular arcs.
+	/// The algorithm can be customized using the [ArcsOptions] structure by setting the following fields within ArcOptions:
+	/// - maximize_arcs - [MaximizeArcs] - The default value is `Automatic`.,
+	/// - error: f64 - The error used for approximating the arc's fit. The default is `0.5`.
+	/// - max_iterations: i32 - The maximum number of segment iterations used as attempts for arc approximations. The default is `100`.
+	pub fn arcs(&self, arcs_options: ArcsOptions) -> Vec<CircleArc> {
+		let ArcsOptions { maximize_arcs, error, max_iterations } = arcs_options;
+		match maximize_arcs {
+			MaximizeArcs::Automatic => {
+				let (auto_arcs, final_low_t) = self.arcs_helper(0., 1., error, max_iterations, true);
+				if final_low_t != 1. {
+					[
+						auto_arcs,
+						self.split(final_low_t)[1].arcs(ArcsOptions {
+							maximize_arcs: MaximizeArcs::Off,
+							error,
+							max_iterations,
+						}),
+					]
+					.concat()
+				} else {
+					auto_arcs
+				}
+			}
+			MaximizeArcs::On => self.arcs_helper(0., 1., error, max_iterations, false).0,
+			MaximizeArcs::Off => self
+				.get_extrema_t_list()
+				.windows(2)
+				.flat_map(|t_pair| self.arcs_helper(t_pair[0], t_pair[1], error, max_iterations, false).0)
+				.collect::<Vec<CircleArc>>(),
+		}
+	}
+
+	/// Implements an algorithm that approximates a bezier curve with circular arcs.
+	/// This algorithm uses a method akin to binary search to find an arc that approximates a maximal segment of the curve.
+	/// Once a maximal arc has been found for a sub-segment of the curve, the algorithm continues by starting again at the end of the previous approximation.
+	/// More details can be found in the [Approximating a Bezier curve with circular arcs](https://pomax.github.io/bezierinfo/#arcapproximation) section of Pomax's bezier curve primer.
+	/// A caveat with this algorithm is that it is possible to find erroneous approximations in cases such as in a very narrow `U`.
+	/// - `stop_when_invalid`: Used to determine whether the algorithm should end early erroneous approximations are encountered.
```

Comment:
After "end early": missing the word "if" or "when". Also maybe change "end" to "terminate".